airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoUtil.java

```diff
@@ -186,7 +186,8 @@ public static Optional<CollectionStatistics> getCollectionStatistics(final Mongo
           final Document stats = cursor.next();
           @SuppressWarnings("unchecked")
           final Map<String, Object> storageStats = (Map<String, Object>) stats.get(MongoConstants.STORAGE_STATS_KEY);
-          if (storageStats != null && !storageStats.isEmpty()) {
+          if (storageStats != null && !storageStats.isEmpty() && storageStats.containsKey(MongoConstants.COLLECTION_STATISTICS_COUNT_KEY)
```

Comment:
can we just abstract this to a `isStorageStats()` valid method

Reply:
oops sorry hit submit before seeing this. I'll keep this in mind and ship this improvement next time i touch mongodb